From 0f13d36cd09b357fd60df355023915ef6cbb1093 Mon Sep 17 00:00:00 2001
From: CI Assistant <chdauto@intel.com>
Date: Mon, 13 Apr 2020 11:10:37 +0300
Subject: [PATCH] Merge pull request #1999 in SW_UGW/linux from
 DRV_DPM_SW-219-fix_wrong_ep_dp_rx_8_4_1 to 8.4.1

* commit 'b24eb2b4b48f7ee4e753bbedb5c9c691e1b466e7':
  DRV_DPM_SW-219 - To fix ep value range checking in dp_rx for GRX(DRVLIB_SW-3542)
---
 drivers/net/datapath/dpm/gswip30/datapath_rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/datapath/dpm/gswip30/datapath_rx.c b/drivers/net/datapath/dpm/gswip30/datapath_rx.c
index 71e9a681d3d5..683c4041f7e8 100644
--- a/drivers/net/datapath/dpm/gswip30/datapath_rx.c
+++ b/drivers/net/datapath/dpm/gswip30/datapath_rx.c
@@ -287,7 +287,7 @@ int32_t dp_rx_30(struct sk_buff *skb, u32 flags)
 	}
 
 	if (unlikely(port_id >=
-	    dp_port_prop[inst].info.cap.max_num_dp_ports - 1)) {
+	    dp_port_prop[inst].info.cap.max_num_dp_ports)) {
 		pr_err("Drop for wrong ep or src port id=%u ??\n",
 		       port_id);
 		goto RX_DROP;
